hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1097,87 +1165,76 @@ protected void cleanIfNecessary(BaseHoodieWriteClient writeClient, String instan
-          .map(p -> {
-            String partitionName = HoodieTableMetadataUtil.getPartitionIdentifier(p.getRelativePath());
-            return Pair.of(partitionName, p.getFileNameToSizeMap());
-          })
-          .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
-
-      int totalDataFilesCount = partitionToFilesMap.values().stream().mapToInt(Map::size).sum();
-      List<String> partitions = new ArrayList<>(partitionToFilesMap.keySet());
-
-      if (partitionTypes.contains(MetadataPartitionType.FILES)) {
-        // Record which saves the list of all partitions
-        HoodieRecord allPartitionRecord = HoodieMetadataPayload.createPartitionListRecord(partitions);
-        HoodieData<HoodieRecord> filesPartitionRecords = getFilesPartitionRecords(createInstantTime, partitionInfoList, allPartitionRecord);
-        ValidationUtils.checkState(filesPartitionRecords.count() == (partitions.size() + 1));
-        partitionToRecordsMap.put(MetadataPartitionType.FILES, filesPartitionRecords);
-      }
-
-      if (partitionTypes.contains(MetadataPartitionType.BLOOM_FILTERS) && totalDataFilesCount > 0) {
-        final HoodieData<HoodieRecord> recordsRDD = HoodieTableMetadataUtil.convertFilesToBloomFilterRecords(
-            engineContext, Collections.emptyMap(), partitionToFilesMap, getRecordsGenerationParams(), createInstantTime);
-        partitionToRecordsMap.put(MetadataPartitionType.BLOOM_FILTERS, recordsRDD);
-      }
-
-      if (partitionTypes.contains(MetadataPartitionType.COLUMN_STATS) && totalDataFilesCount > 0) {
-        final HoodieData<HoodieRecord> recordsRDD = HoodieTableMetadataUtil.convertFilesToColumnStatsRecords(
-            engineContext, Collections.emptyMap(), partitionToFilesMap, getRecordsGenerationParams());
-        partitionToRecordsMap.put(MetadataPartitionType.COLUMN_STATS, recordsRDD);
-      }
-      LOG.info("Committing " + partitions.size() + " partitions and " + totalDataFilesCount + " files to metadata");
+  private boolean validateTimelineBeforeSchedulingCompaction(Option<String> inFlightInstantTimestamp, String latestDeltacommitTime) {
+    // There should not be any incomplete instants on MDT
+    HoodieActiveTimeline metadataTimeline = metadataMetaClient.reloadActiveTimeline();
+    List<HoodieInstant> pendingInstantsOnMetadataTable = metadataTimeline.filterInflightsAndRequested().getInstants();
+    if (!pendingInstantsOnMetadataTable.isEmpty()) {
+      LOG.info(String.format(
+              "Cannot compact MDT as there are %d inflight instants: %s",
+              pendingInstantsOnMetadataTable.size(), Arrays.toString(pendingInstantsOnMetadataTable.toArray())));
+      metrics.ifPresent(m -> m.incrementMetric(HoodieMetadataMetrics.SKIP_TABLE_SERVICES, 1));
+      return false;
     }
 
-    commit(createInstantTime, partitionToRecordsMap, false);
-  }
+    // There should not be any incomplete instants on dataset
```

Comment:
I guess the conditions in L1215 - 1234 should subsume this condition (L 1187 to 1197). infact we had it this way probably 2 releases back and evolved it. 
we can probably remove L1215 to 1234.

Reply:
same here

Reply:
Yes, removed duplicate checks.